internal/controller/group_controller.go

```
@@ -327,13 +327,24 @@ func (r *GroupReconciler) processAllBackends(
 	uniqueMembers []string,
 ) map[string]map[string]string {
 	backendErrors := make(map[string]map[string]string, 0)
+
+	// Group Params by backend name for direct lookup.
+	groupParamsByBackend := make(map[string]structs.TeamParams)
+	for _, param := range groupCR.Spec.GroupParams {
+		groupParamsByBackend[param.Name+"_"+param.Backend] = structs.TeamParams{
```

Should we validate that param.Name and param.Backend match an actual backend in groupCR.Spec.Backends? Currently, invalid params are silently ignored.

---

Done